### DISCUSSION TITLE
Is it possible to add a lack of double or simple quotation signs?

### DISCUSSION THREAD

**Discussion by MeatWant** [PHP-CS-Fixer/PHP-CS-Fixer#7883] [Q&A]
Hi everyone,

i searched for a rule to fix very old code of a project with over 2000 files

I have a lot of files in which array variables are not masked with quotation marks.

Can i fix it with php-CS fixer?

Before: $test[noquotation]
After: $test["noquotation"]

I can't find something to do this.

Thanks for any suggest, how to can do this with php-cs-fixer.

LG, Stefan

**Comment by Wirone** [comment]
No, there's no such fixer available. This kind of code is technically valid because `bar` in `$foo[bar]` can be a constant. Fixer does not have such context, and even though this could be done automatically, it would be risky and is such an edge case that IMHO is not worth implementing. Especially that you can do it automatically with a regex substitution in IDE:

<img width="657" alt="image" src="https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/600668/39aeeb87-fb01-4b28-a0b9-4921a3a01e89">

Just replace (<kbd>cmd</kbd>+<kbd>R</kbd>) `\[(\h?)([a-zA-Z0-9_]+)(\h?)\]` with `[$1'$2'$3]` and you're done.
